### PR TITLE
Fix hue rotation and parse as float

### PR DESCRIPTION
**Pull request by tusharpm**
A potential fix for #18

### CODE REVIEW

File: color-string.js [hunk 1]
```diff
@@ -84,11 +84,11 @@ function getHsla(string) {
    if (!string) {
       return;
    }
-   var hsl = /^hsla?\(\s*([+-]?\d+)(?:deg)?\s*,\s*([+-]?[\d\.]+)%\s*,\s*([+-]?[\d\.]+)%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)/;
+   var hsl = /^hsla?\(\s*([+-]?\d+(?:\.\d*)?)(?:deg)?\s*,\s*([+-]?[\d\.]+)%\s*,\s*([+-]?[\d\.]+)%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)/;
```
Comment (Qix-) [inline color-string.js]: > zero or more decimal digits followed by a dot (.)

Does this account for floats like `.5`? Can you add add a test for it and make sure that's what's happening?

**Comment by Qix-** [comment]
Looks good other than that one issue regarding the regex.

File: color-string.js (hunk 1 above)
Comment (tusharpm) [inline color-string.js]: It most likely wouldn't handle such numbers. I can modify the regex to make it similar to the ones being used for the other parameters like `[\d\.]+`. That of course brings in its own flaws.
Alternatively I could use a regex like `\d*[\.]?\d+` which would still not be able to handle exponent notations.
I will update the PR with the second option, as it seems to cover more cases.

**Comment by Qix-** [comment]
Landed in 8627b8aeb1c3c46cef2d3956f9f524710f820abb.

Thank you for the PR!

**Comment by tusharpm** [comment]
Happy to help.
There's one other thing that I just noticed: the README.md file is not updated for the hue rotation change. It still shows wrong expected return values for `getHsl` and `getHsla`. For both the examples there, the hue (the first number in the array) would be rotated from 360 to 0.

**Comment by Qix-** [comment]
@tusharpm mind opening a PR about it? :)

**Pull request closed without merging**